strawberryfields/ops.py

```diff
@@ -542,7 +542,12 @@ def __init__(self, a=0.0, p=0.0):
 
     def _apply(self, reg, backend, **kwargs):
         p = self.p[0] * pf.exp(1j * self.p[1])
-        z = par_evaluate(p)
+
+        if np.any(self.p[1] != 0):
+            z = par_evaluate(p, dtype=np.complex128)
```

Comment:
We should set this dtype once in a single spot, rather than independently in each function

Reply:
done!